### ISSUE TITLE
[CLOSED] Ideas

### ISSUE THREAD

**Issue by bwwagner** [ephemerant/445-Final#1]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ephemerant](https://github.com/ephemerant)**
_Wednesday Aug 24, 2016 at 00:44 GMT_
_Originally opened as https://github.com/ephemerant/445/issues/1_

----

**Comment by bwwagner** [comment]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ephemerant](https://github.com/ephemerant)**
_Wednesday Aug 24, 2016 at 00:50 GMT_

----

Speak with Dr. Lang?

**Comment by bwwagner** [comment]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ephemerant](https://github.com/ephemerant)**
_Wednesday Aug 24, 2016 at 00:59 GMT_

----

Potential idea with mathematical visualizations of topics to assist learning

**Comment by bwwagner** [comment]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ephemerant](https://github.com/ephemerant)**
_Wednesday Aug 24, 2016 at 01:00 GMT_

----

Potential AI ideas?

**Comment by bwwagner** [comment]
<a href="https://github.com/jrlytle"><img src="https://avatars5.githubusercontent.com/u/14928183?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jrlytle](https://github.com/jrlytle)**
_Wednesday Aug 24, 2016 at 13:24 GMT_

----

Just sent an email to Dr. Lang. Waiting on reply, will post again as soon as I hear back.

**Comment by bwwagner** [comment]
<a href="https://github.com/bwwagner"><img src="https://avatars4.githubusercontent.com/u/13163930?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [bwwagner](https://github.com/bwwagner)**
_Thursday Aug 25, 2016 at 18:40 GMT_

----

I looked at Mathematica a bit, and it seems there is a specialized programming language which Wolfram uses to interface with it. It is known simply as [the Wolframn language (http://reference.wolfram.com/language/.](http://reference.wolfram.com/language)).

If we incorporate Mathematica we may end up dealing with licensing and grant proposals. [Here is a link to API info for Mathematica](https://reference.wolfram.com/language/guide/CreatingAnInstantAPI.html)

**Comment by bwwagner** [comment]
<a href="https://github.com/jrlytle"><img src="https://avatars5.githubusercontent.com/u/14928183?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jrlytle](https://github.com/jrlytle)**
_Thursday Aug 25, 2016 at 19:36 GMT_

----

If it looks like it would be too much trouble, then that's fine because I still haven't received a reply from Dr. Lang.

**Comment by bwwagner** [comment]
<a href="https://github.com/jrlytle"><img src="https://avatars5.githubusercontent.com/u/14928183?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jrlytle](https://github.com/jrlytle)**
_Thursday Aug 25, 2016 at 19:40 GMT_

----

Forget what I just said, he just replied to my message. Here's his email:

Jason, 

I’m not sure if I can give a good answer to your question. I’m thinking if your focus is on the linear algebra itself—programming the algorithms for matrix multiplication and so on—that might give you some experience with numerical analysis and linear algebra. But I’m not sure that would be interesting or valuable for you, and there are already programs and libraries for linear algebra that are widely used and powerful. What I think might be more interesting would be a program that was more tutorial in nature: a user interface that would allow students to specify row operations for solving problems with matrices. 

There are existing programs such as MATLAB or OCTAVE that allow the user to perform row operations on a matrix, as well as do such things as find the inverse of a matrix or its eigenvalues and eigenvectors. But these programs just have a command-line console as an interface, which makes them hard to use. (MATLAB and OCTAVE are really programming languages, not unlike R or Python.) Perhaps a program that shows a matrix (or a system of equations) and allows the user to select operations from a list, and shows the updated matrix? 

One possible challenge: when you do this by hand, you’re probably using fractions not decimals. So you might get x = 4/7 not x = 0.5714. It might be nice to have a program that lets the user do rational numbers instead of decimal approximations. But exact answers wouldn’t be feasible for some basic linear algebra operations, such as finding eigenvalues. If you decide to program that sort of thing, you might look for existing linear algebra libraries (LAPACK) and figure out how to link them to your code and call the functions they contain. That might be an interesting task in a computer science nuts-and-bolts sort of way. (LAPACK is written in Fortran but packages like the language R use it.) 

Hope this helps, but best wishes for your project. 

Dr. Lang

**Comment by bwwagner** [comment]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ephemerant](https://github.com/ephemerant)**
_Sunday Aug 28, 2016 at 17:33 GMT_

----

That moment when he still hasn't posted the project ideas...

**Comment by bwwagner** [comment]
<a href="https://github.com/jrlytle"><img src="https://avatars5.githubusercontent.com/u/14928183?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jrlytle](https://github.com/jrlytle)**
_Sunday Aug 28, 2016 at 21:30 GMT_

----

They'll probably be posted about 5 min before class starts on Tuesday...along with the schedule of class meetings.

**Comment by bwwagner** [comment]
<a href="https://github.com/bwwagner"><img src="https://avatars4.githubusercontent.com/u/13163930?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [bwwagner](https://github.com/bwwagner)**
_Monday Aug 29, 2016 at 14:42 GMT_

----

I'll share some of my brainstorming for the project

Class/Math Related
- Virtual Teacher's Assistant/Aid (think repurposed Siri)
- Class App (could include ebooks, calculator, testing)
- Math Grading using handwriting recognition (abstraction of stylus reading
  algorithms)

I have other ideas, but they aren't as closely related to the topic.

**Comment by bwwagner** [comment]
<a href="https://github.com/ephemerant"><img src="https://avatars7.githubusercontent.com/u/9091188?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ephemerant](https://github.com/ephemerant)**
_Monday Aug 29, 2016 at 15:17 GMT_

----

Post any and all ideas - may the best topic win!

**Comment by bwwagner** [comment]
<a href="https://github.com/bwwagner"><img src="https://avatars4.githubusercontent.com/u/13163930?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [bwwagner](https://github.com/bwwagner)**
_Tuesday Aug 30, 2016 at 23:17 GMT_

----

Not all of these are feasible, please consider these as brainstorming.

Marketing
- Spoofing online signature to study marketing collection practices
- Studying the effects of a compromised electronic signature

Behavior
- Why do people cyber-bully a study the treatment of game avatars in the Sims
- Examine the effects of anonymity on human interactions

Engineering
-  Creation of instantaneous networks using unmanned drones as mobile hotspots
-  Creation of bit streams using multiple static files
